src/views/datasources/actions/DeleteDataSourceModal/hooks/useUnderlyingPVC.ts

```diff
@@ -0,0 +1,36 @@
+import {
+  modelToGroupVersionKind,
+  PersistentVolumeClaimModel,
+} from '@kubevirt-ui/kubevirt-api/console';
+import DataVolumeModel from '@kubevirt-ui/kubevirt-api/console/models/DataVolumeModel';
+import {
+  V1beta1DataSource,
+  V1beta1DataVolume,
+} from '@kubevirt-ui/kubevirt-api/containerized-data-importer/models';
+import { IoK8sApiCoreV1PersistentVolumeClaim } from '@kubevirt-ui/kubevirt-api/kubernetes';
+import { useK8sWatchResource } from '@openshift-console/dynamic-plugin-sdk';
+
+type UseUnderlyingPVC = (dataSource: V1beta1DataSource) => {
+  dv: V1beta1DataVolume;
+  pvc: IoK8sApiCoreV1PersistentVolumeClaim;
+};
+
+const useUnderlyingPVC: UseUnderlyingPVC = (dataSource) => {
+  const name = dataSource?.spec?.source?.pvc?.name;
+  const namespace = dataSource?.spec?.source?.pvc?.namespace;
+  const [pvc, pvcLoaded] = useK8sWatchResource<IoK8sApiCoreV1PersistentVolumeClaim>({
+    groupVersionKind: modelToGroupVersionKind(PersistentVolumeClaimModel),
+    name,
+    namespace,
+  });
+
+  const [dv, dvLoaded] = useK8sWatchResource<V1beta1DataVolume>({
+    groupVersionKind: modelToGroupVersionKind(DataVolumeModel),
+    name,
+    namespace,
+  });
+
+  return { dv: dvLoaded && dv, pvc: pvcLoaded && pvc };
```

Comment:
what about default data? if someone expect dv and pvc to get an object it will return false

Reply:
added a check to make sure the objects aren't empty

Reply:
better do this check in the hook and not in the consuming component, instead of returning false, better to return empty object and loaded boolean (for loaders)